broker/aws/impl.go

```diff
@@ -235,19 +235,24 @@ func (b *Broker) getUserAllowedAccountsFromGroups(userGroups []string) ([]broker
 		groupToAccountName[groupName] = account.Name
 		groupList = append(groupList, groupName)
 	}
-	reString := fmt.Sprintf("(?i)(%s)-(.*)$", strings.Join(groupList, "|"))
-	re, err := regexp.Compile(reString)
-	if err != nil {
-		return nil, err
-	}
 	var allowedRoles map[string][]string
 	allowedRoles = make(map[string][]string)
-	for _, group := range userGroups {
-		matches := re.FindStringSubmatch(group)
-		if len(matches) == 3 {
-			b.logger.Debugf(2, "matches=%v", matches)
-			accountGroupName := strings.ToLower(matches[1])
-			allowedRoles[accountGroupName] = append(allowedRoles[accountGroupName], matches[2])
+
```

Comment:
// Previously we were doing account matching using a single regexp.
        // However, that mechanism can yield to an extension attack when accounts
        // have the same prefix. Now we explicitly build a single refexp per account